File: PYTHON/Pertemuan 05/Mahasiswa.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class Mahasiswa:
    # Menggunakan default arguments untuk mensimulasikan overloading konstruktor di Python
    def __init__(self, nim: str, nama: str, program_studi: str = "Sistem Informasi", tahun_masuk: int = 2025, ipk: float = 0.0):
        self._nim = nim
        self._nama = nama
        self._program_studi = program_studi
        self._tahun_masuk = tahun_masuk
        self._ipk = ipk

    def set_ipk(self, ipk: float):
        if 0.0 <= ipk <= 4.0:
            self._ipk = ipk

    def tampilkan(self):
        print(f"  {self._nim:<12} | {self._nama:<20} | {self._program_studi:<20} | {self._tahun_masuk} | IPK: {self._ipk:.2f}")

File: JAVA/Pertemuan 05/src/id/ac/uph/pbo/pertemuan05/Mahasiswa.java
package id.ac.uph.pbo.pertemuan05;

public class Mahasiswa {
    private String nama;
    private String nim;
    private String programStudi;
    private int    tahunMasuk;
    private double ipk;

    // Nilai default Java: 0, 0.0, false, null (jika tidak di-set)

    // Konstruktor 1: minimal (hanya nim & nama)
    Mahasiswa(String nim, String nama) {
        this(nim, nama, "Sistem Informasi", 2025, 0.0);
    }

    // Konstruktor 2: dengan program studi
    Mahasiswa(String nim, String nama, String programStudi) {
        this(nim, nama, programStudi, 2025, 0.0);
    }

    // Konstruktor 3: lengkap
    Mahasiswa(String nim, String nama, String programStudi, int tahunMasuk, double ipk) {
        this.nim          = nim;
        this.nama         = nama;
        this.programStudi = programStudi;
        this.tahunMasuk   = tahunMasuk;
        this.ipk          = ipk;
    }

    void setIpk(double ipk) {
        if (ipk >= 0.0 && ipk <= 4.0) this.ipk = ipk;
    }

    void tampilkan() {
        System.out.printf("  %-12s | %-20s | %-20s | %d | IPK: %.2f%n",
                nim, nama, programStudi, tahunMasuk, ipk);
    }
}
